test/SConstruct

```diff
@@ -18,29 +18,29 @@ if env["target"] in ["editor", "template_debug"]:
     doc_data = env.GodotCPPDocData("src/gen/doc_data.gen.cpp", source=Glob("doc_classes/*.xml"))
     sources.append(doc_data)
 
-if env["platform"] == "macos":
-    library = env.SharedLibrary(
-        "project/bin/libgdexample.{}.{}.framework/libgdexample.{}.{}".format(
-            env["platform"], env["target"], env["platform"], env["target"]
-        ),
+if env["platform"] == "macos" or env["platform"] == "ios":
+    # Static libraries (.dylib) are not supported on the iOS app store.
+    # For consistency, both macOS and iOS will generate .xcframework instead.
+    framework_tool = Tool("apple_framework", toolpath=["../tools"])
+
+    framework_name = f"gdexample.{env['platform']}.{env['target']}"
+    if env["ios_simulator"]:
+        framework_name += ".simulator"
```

Comment:
Why? The only purpose of `.xcframework` is to contain libs for multiple targets, so there should not be separate simulator xcframework.

Reply:
I agree, the previous code confused me too, but I merely adapted what I found.